lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -64,6 +64,20 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
             // NodeOp to add the InnerSym.
             if (!xmrDef.isa<BlockArgument>()) {
               Operation *xmrDefOp = xmrDef.getDefiningOp();
+              if (auto verbExpr = dyn_cast<VerbatimExprOp>(xmrDefOp))
+                if (verbExpr.getSymbolsAttr().empty() &&
+                    xmrDefOp->hasOneUse()) {
+                  // This represents the internal path into a module. For
+                  // generating the correct XMR, no node can be created in this
+                  // module. Create a null InnerRef and ensure the hierarchical
+                  // path ends at the parent that instantiates this module.
+                  auto inRef = InnerRefAttr();
+                  auto ind = addReachingSendsEntry(send.getResult(), inRef);
+                  xmrPathSuffix[ind] = ("." + verbExpr.getText()).str();
+                  markForRemoval(verbExpr);
+                  markForRemoval(send);
+                  return success();
+                }
```

Comment:
What should happen here if the condition doesn't hold? Or: is there a reason to gate this based on the symbol being empty? I think it's just a reason to not mark the `verbExpr` for removal.

Reply:
And also, we currently are not handling the symbol substitution for suffix. It should be simple to update the data structures to keep track of the symbols also. 
So, the condition is narrowly checking for a verbatim with an opaque "internal path", that would be generated by the `internalPath` annotation. 
You are right, if I keep track of the symbols for suffix substitution, the check is not required, other than marking for removal.